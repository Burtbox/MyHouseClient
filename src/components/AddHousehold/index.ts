import AddHousehold, { AddHousehold as DisconnectedAddHousehold } from './AddHousehold';

export { DisconnectedAddHousehold };
export default AddHousehold;
