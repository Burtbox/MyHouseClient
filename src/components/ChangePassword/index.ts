import ChangePassword, { ChangePassword as DisconnectedChangePassword } from './ChangePassword';

export default ChangePassword;

export { DisconnectedChangePassword };
