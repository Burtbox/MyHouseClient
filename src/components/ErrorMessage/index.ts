import ErrorMessage, { ErrorMessage as DisconnectedErrorMessage } from './ErrorMessage';

export { DisconnectedErrorMessage };
export default ErrorMessage;
