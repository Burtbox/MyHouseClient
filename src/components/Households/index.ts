import Households, { Households as DisconnectedHouseholds } from './Households';

export default Households;

export { DisconnectedHouseholds };
