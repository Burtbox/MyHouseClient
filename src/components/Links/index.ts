import Links, { Links as DisconnectedLinks } from './Links';

export default Links;
export { DisconnectedLinks };
