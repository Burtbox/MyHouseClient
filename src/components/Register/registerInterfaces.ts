import { IComponentProps } from '../../interfaces/componentInterfaces';
import { IFormStyles } from '../../styles/styles';
import { IUserAuthenticationObject } from '../Users/usersInterfaces';

export interface IRegisterProps extends IRegisterReducer, IComponentProps, IFormStyles { }

export interface IRegisterState {
    registerUser: {
        email: string,
        password: string,
        confirmPassword: string,
        displayName: string,
    };
}

export interface IRegisterUserObject extends IUserAuthenticationObject {
    displayName: string;
    confirmPassword: string;
}

export interface IRegisterReducer {
    registering: boolean;
}
