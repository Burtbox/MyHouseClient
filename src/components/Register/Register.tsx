import { Typography } from '@material-ui/core';
import { default as Button } from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import * as React from 'react';
import { connect } from 'react-redux';
import { myHouseRoutes } from '../../enums/routesEnum';
import { IStore } from '../../interfaces/storeInterface';
import appStyles from '../../styles';
import { ErrorMessageActions } from '../ErrorMessage/errorMessageActions';
import Loading from '../Loading';
import { registerUser } from './registerEpic';
import { IRegisterProps, IRegisterState, IRegisterUserObject } from './registerInterfaces';

export class Register extends React.Component<IRegisterProps, IRegisterState> {
    constructor(props: IRegisterProps) {
        super(props);
        this.state = {
            registerUser: {
                email: '',
                password: '',
                confirmPassword: '',
                displayName: '',
            },
        };
    }

    handleAddUser = (event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        if (this.state.registerUser.password === this.state.registerUser.confirmPassword) {

            const user: IRegisterUserObject = {
                displayName: this.state.registerUser.displayName,
                email: this.state.registerUser.email,
                password: this.state.registerUser.password,
                confirmPassword: this.state.registerUser.confirmPassword,
            };
            registerUser(user);
        } else {
            this.props.dispatch(ErrorMessageActions.addError('Your password does not match the confirmed password'));
        }
    }

    handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
        const target = event.target;
        const value = target.value;
        const name = target.name;

        this.setState(prevState => ({
            registerUser: { ...this.state.registerUser, [name]: value },
        }));
    }

    render() {
        return (
            <form style={appStyles.container} onSubmit={this.handleAddUser}>
                <Typography variant="headline">Create Account</Typography>
                <div>
                    <span style={{ verticalAlign: 'middle' }}>
                        Already have a My House account?
              </span> {' '}
                    <Button
                        variant="outlined"
                        onClick={() => this.props.history.push(myHouseRoutes.Login)}
                    >
                        Sign In
                    </Button>
                </div>

                <div>
                    <TextField
                        name="displayName"
                        placeholder="My name"
                        label="Display Name"
                        required
                        onChange={this.handleInputChange}
                        disabled={this.props.registering}
                        margin="normal"
                    />
                </div>
                <div>
                    <TextField
                        name="email"
                        placeholder="example@email.com"
                        label="Email Address"
                        required
                        onChange={this.handleInputChange}
                        disabled={this.props.registering}
                        margin="normal"
                    />
                </div>
                <div>
                    <TextField
                        name="password"
                        type="password"
                        placeholder="**********"
                        label="Password"
                        required
                        onChange={this.handleInputChange}
                        disabled={this.props.registering}
                        margin="normal"
                    />
                </div>
                <div>
                    <TextField
                        name="confirmPassword"
                        type="password"
                        placeholder="**********"
                        label="Confirm Password"
                        required
                        onChange={this.handleInputChange}
                        disabled={this.props.registering}
                        margin="normal"
                    />
                </div>
                <div>
                    {this.props.registering ? (
                        <Loading />
                    ) : (
                            <Button type="submit" >
                                Sign Up
                            </Button>
                        )}
                </div>
            </form>
        );
    }
}

const mapStateToProps = (store: IStore) => {
    return {
        registering: store.registerReducer.registering,
    };
};

export default connect(mapStateToProps)(Register);
