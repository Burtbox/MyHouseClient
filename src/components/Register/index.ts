import Register, { Register as DisconnectedRegister } from './Register';

export default Register;

export { DisconnectedRegister };
