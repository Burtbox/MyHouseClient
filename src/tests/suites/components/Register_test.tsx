import * as React from 'react';
import { shallow } from 'enzyme';
import { DisconnectedRegister } from '../../../components/Register';
import { IRegisterProps } from '../../../components/Register/registerInterfaces';

describe('Register test suite', () => {
    test('Render register screen', () => {
        const props: Readonly<IRegisterProps> = {
            history: undefined,
            registering: false,
            dispatch: undefined,
        };
        const registerScreen = shallow(
            <DisconnectedRegister {...props} />,
        );
        expect(registerScreen).toMatchSnapshot();
    });
});
