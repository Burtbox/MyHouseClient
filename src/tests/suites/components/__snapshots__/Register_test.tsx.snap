// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register test suite Render register screen 1`] = `
<form
  onSubmit={[Function]}
  style={
    Object {
      "marginTop": "20px",
      "textAlign": "center",
    }
  }
>
  <h2>
    Create Account
  </h2>
  <div>
    <span
      style={
        Object {
          "vertical-align": "middle",
        }
      }
    >
      Already have a My House account?
    </span>
     
     
    <FlatButton
      disabled={false}
      fullWidth={false}
      label="Sign In"
      labelPosition="after"
      labelStyle={Object {}}
      onClick={[Function]}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={true}
    />
  </div>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Display Name"
      fullWidth={false}
      hintText="My name"
      maxlength="100"
      multiLine={false}
      name="displayName"
      onChange={[Function]}
      required={true}
      rows={1}
      type="text"
      underlineShow={true}
    />
  </div>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Email Address"
      fullWidth={false}
      hintText="example@email.com"
      maxlength="50"
      multiLine={false}
      name="email"
      onChange={[Function]}
      required={true}
      rows={1}
      type="text"
      underlineShow={true}
    />
  </div>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Password"
      fullWidth={false}
      hintText="**********"
      maxlength="30"
      multiLine={false}
      name="password"
      onChange={[Function]}
      required={true}
      rows={1}
      type="password"
      underlineShow={true}
    />
  </div>
  <div>
    <TextField
      disabled={false}
      floatingLabelFixed={false}
      floatingLabelText="Confirm Password"
      fullWidth={false}
      hintText="**********"
      maxlength="30"
      multiLine={false}
      name="confirmPassword"
      onChange={[Function]}
      required={true}
      rows={1}
      type="password"
      underlineShow={true}
    />
  </div>
  <div>
    <FlatButton
      disabled={false}
      fullWidth={false}
      label="Sign Up"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      type="submit"
    />
  </div>
</form>
`;
